src/stdlib/dns_lookup.rs

```diff
@@ -12,10 +15,69 @@ mod non_wasm {
     use domain::resolv::stub::conf::{ResolvConf, ResolvOptions, ServerConf, Transport};
     use domain::resolv::stub::Answer;
     use domain::resolv::StubResolver;
+    use once_cell::sync::Lazy;
+    use tokio::runtime::Handle;
 
     use crate::compiler::prelude::*;
     use crate::value::Value;
 
+    static WORKER: Lazy<Worker> = Lazy::new(Worker::new);
+
+    type Job<T> = Box<dyn FnOnce() -> T + Send + 'static>;
+    struct JobHandle<T> {
+        job: Job<T>,
+        result: Arc<mpsc::Sender<T>>,
+    }
+
+    struct Worker {
+        thread: Option<thread::JoinHandle<()>>,
+        queue: Option<mpsc::Sender<JobHandle<Result<Answer, Error>>>>,
+    }
+
+    impl Worker {
```

Comment:
Did you borrow this implementation from somewhere? It's slightly more complicated than I might have expected.

Reply:
It is roughly based on https://doc.rust-lang.org/book/ch20-02-multithreaded.html
I tried to simplify it as much as I could, but I thought I had to have some kind of channel to take in the function and to send back the result.

Reply:
Ah, gotcha, yeah I see. It seems like that example just creates one channel that is reused where this implementation creates a channel per call to `execute`.  Is there a reason we need to create one channel per `execute` call?

Reply:
Oh, right. My bad, one should be enough.

Reply:
Right, a single `mpsc::sync_channel(CHANNEL_CAPACITY)` should be enough.
Both the sender and the receiver should be lazily instantiated instances.

Reply:
It is all contained in `Worker` instance now, so both of them are lazily insttantiated (since `Worker` is lazily instantiated). There are 2 bounded channels though, one for jobs and one for results. I hope that is alright.

I have made the capacity 0 for testing (meaning it is always blocking), but on the other hand, not sure if it makes sense to make it any bigger, considering there is only 1 thread handling jobs. I might be missing something,